config_utilities/demos/demo_config.cpp

```diff
@@ -160,78 +161,362 @@ void declare_config(SubSubConfig& config) {
+  //this relies on the type index being the same, maybe worth doing some actual casting to check
+  std::map<std::type_index, std::shared_ptr<ReconfigureInterface>> registered_interface_;
 
-  // Any errors parsing configs will print verbose warnings if desired and use the default values.
-  invalid_config = config::fromYamlFile<demo::MyConfig>(my_root_path + "invalid_params.yaml");
+};
 
-  // ======================================== Printing configs to string ========================================
-  std::cout << "\n\n----- Printing configs to string -----\n\n" << std::endl;
+int main(int argc, char** argv) {
+  if (argc < 2) {
+    std::cerr << "invalid usage! expected resource directory as argument" << std::endl;
+    return 1;
+  }
 
-  // Easier automatic printing of all configs with unit and additional information can be done using the toString():
-  const std::string config_as_string = config::toString(config);
-  std::cout << config_as_string << std::endl;
+  const std::string my_root_path = std::string(argv[1]) + "/";
 
-  // Inclunding "printing.h" also implements the ostream operator for decared config types. The above is thus equivalent
-  // to:
-  std::cout << config << std::endl;
+  // GLobal settings can be set at runtime to change the behavior and presentation of configs.
+  config::Settings().inline_subconfig_field_names = true;
+  config::Settings().print_warnings = true;
+
+  SimpleConfig config;
+  std::cout << config::toString(config) << std::endl;
+  auto interface = std::make_shared<ManualInterface>();
+  DynamicReconfigureServer::registerInterface(interface);
+
+  //I am unclear about this key. Does it need to be namespaced according to anything else in the system?
+  //Ideally not as its just some key that refers to an INSTANCE of a config and not the Config type (like config::name())
+  //some bookeeping could be done here. 
+  DynamicReconfigureServer::registerConfig<ManualInterface>(&config, "my_config");
+
+  SimpleConfig new_config;
+  interface->set(new_config, "my_config");
+
+  new_config.f = 10.0;
+  interface->set(new_config, "my_config");
+
+  std::cout << config::toString(config) << std::endl;
+
+  interface->set(new_config, "non_existant_config");
+
+  //should not work as different config types (between what is registered as "my_config" and the incoming config)
+  demo::SubConfig sub_config;
+  interface->set(sub_config, "my_config");
+
+
+  /**
+   * @brief Ideas
+   * sub config modification via namespacing
+   * a config is registered with a name, e.g "my_config" and (using the MetaData) we make namespacing for each of the fields
+   * e.g my_config/f, my_config/s
+   * 
+   * The interface gets a new function that allows sub-updating e.g. Interface::setField(T, namespace)
+   * e.g interface->setField(10.1, "my_config/f")
+   * 
+   * since we have access to my_config and we know how to access the YAML field 'f' associated with that config we can just update the field in the yaml.
```

Comment:
I don't think that will quite work since constructing the SimpleConfig will set the other fields to default, but calling update on the existing config (or a copy) will do what you want.